src/compute/src/lib.rs

```diff
@@ -28,98 +28,121 @@ pub mod memory_management;
 pub mod rpc;
 pub mod server;
 
-use clap::clap_derive::ArgEnum;
 use clap::Parser;
+use risingwave_common::config::{
+    load_config, AsyncStackTraceOption, ComputeNodeConfig, OverwriteConfig, RwConfig,
+};
 use risingwave_common::util::resource_util::cpu::total_cpu_available;
 use risingwave_common::util::resource_util::memory::total_memory_available_bytes;
 
-#[derive(Debug, Clone, ArgEnum)]
-pub enum AsyncStackTraceOption {
-    Off,
-    On, // default
-    Verbose,
-}
-
-/// Command-line arguments for compute-node.
+/// CLI arguments received by compute node. Overwrites fields in
+/// [`risingwave_common::config::ComputeNodeConfig`].
 #[derive(Parser, Clone, Debug)]
 pub struct ComputeNodeOpts {
     // TODO: rename to listen_address and separate out the port.
-    #[clap(long, default_value = "127.0.0.1:5688")]
-    pub host: String,
+    #[clap(long = "host")]
+    pub listen_addr: Option<String>,
```

Comment:
This would break backward compatibility. It seems a little bit troublesome to handle for the cloud side. Need to confirm with them. cc. @arkbriar

Reply:
`long = "host"` means the cli arg will still be `host`.

Reply:
This is already broken in #7527 and #7530 🫣

https://risingwave-labs.slack.com/archives/C0358PVT2AX/p1674626641010649

Reply:
I guess we should put those alternatives back similar to here. However, for meta, it is quite confused: there is both a listen_addr and a host.

Reply:
@fuyufjh it seems the `long = "host"` keeps the compatibility. However, I do think `host` is a bad name :)

Reply:
I'm thinking if we really need two addresses here. By specifying `listen-addr`, we're acting as a server and telling other components how to reach us. When we're acting as a client, we can actually randomly bind to an address just like a web browser does.

Reply:
> When we're acting as a client, we can actually randomly bind to an address just like a web browser does.

For client address (to be registered at meta service), we will first try to use `client_address`. If it is `None`, we fallback to `listen_addr`.

Reply:
IIRC, it seems we're only using this address for an identifier. 🤔

Reply:
> Randomly bind to an address seems ok.

Maybe not 🤣. Different nodes may bind to the same address.

Reply:
> > Randomly bind to an address seems ok.
> 
> 
> 
> Maybe not 🤣. Different nodes may bind to the same address.

We're asking the operating system to allocate a port for us, if we create a socket without specifying a binding address.

Reply:
Right, but what if there are multiple machines...